File: C/pagamento.c
#include <stdio.h>
#include <string.h>

void ler_texto(char *buffer, int length) {
fgets(buffer, length, stdin);
strtok(buffer, "\n");
}

int main() {

    char nome[50];
    double valorHora, pagamento;
    int horasTrabalhadas;

    printf("Nome: ");
    ler_texto(nome, 50);
    printf("Valor por hora: ");
    scanf("%lf", &valorHora);
    printf("Horas trabalhadas: ");
    scanf("%d", &horasTrabalhadas);

    pagamento = valorHora * horasTrabalhadas;

    printf("O pagamento para %s deve ser %.2lf", nome, pagamento);

    return 0;
}

File: Java/pagamento.java
import java.util.Locale;
import java.util.Scanner;

public class Pagamento {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        String nome;
        double valorHora, pagamento;
        int horasTrabalhadas;

        System.out.print("Nome: ");
        nome = sc.nextLine();
        System.out.print("Valor por hora: ");
        valorHora = sc.nextDouble();
        System.out.print("Horas trabalhadas: ");
        horasTrabalhadas = sc.nextInt();

        pagamento = valorHora * horasTrabalhadas;

        System.out.print("O pagamento para " + nome + " deve ser " + String.format("%.2f",pagamento));

        sc.close();
    }
}